www/js/controller.js

```diff
@@ -285,14 +273,147 @@ define([
+					if(this.genericDivMap[directive.id]){
+						domStyle.set(this.genericDivMap[directive.id], directive.attribute, directive.value);
+					}
+				}else{
+					//this code needs to be uncommented when logging module is included
+					/*
+					this.logging.clientLog("warning", {
+						message: "Directive with unknown id, id :"+directive.id,
+						functionTag: 'applyDirectives'
+					});
+					*/
+				}
+			}, this);
+			/* check after node editor tour has been included into topomath
+			if(tempDirective && this.activityConfig.get("showNodeEditorTour")) {
+				this.continueTour(tempDirective);
+			}
+			*/
+		},
+
+		updateModelStatus: function(desc){
+			//stub for updateModelStatus
+			//actual implementation in con-student and con-author
+		},
+		plusHandler: function(){
+			console.log("****** plus button");
+			this.equationInsert('+');
+		},
+		minusHandler: function(){
+			console.log("****** minus button");
+			this.equationInsert('-');
+		},
+		timesHandler: function(){
+			console.log("****** times button");
+			this.equationInsert('*');
+		},
+		divideHandler: function(){
+			console.log("****** divide button");
+			this.equationInsert('/');
+		},
+		equationInsert: function(text){
+			//todo: initialize domNode
+			var widget = registry.byId(this.controlMap.equation);
+			var oldEqn = widget.get("value");
+			// Get current cursor position or go to end of input
+			// console.log("	   Got offsets, length: ", widget.domNode.selectionStart, widget.domNode.selectionEnd, oldEqn.length);
+			var p1 = widget.domNode.selectionStart;
+			var p2 = widget.domNode.selectionEnd;
+			widget.set("value", oldEqn.substr(0, p1) + text + oldEqn.substr(p2));
+			// Set cursor to end of current paste
+			widget.domNode.selectionStart = widget.domNode.selectionEnd = p1 + text.length;
+		},
+
+		updateNodes: function(){
+			// Update node editor and the model.
+			this._nodeEditor.set('title', this._model.active.getName(this.currentID));
 
+			// Update inputs and other equations based on new quantity.
+			expression.addQuantity(this.currentID, this._model.active);
+
+			// need to delete all existing outgoing connections
+			// need to add connections based on new inputs in model.
+			// add hook so we can do this in draw-model...
+			this.addQuantity(this.currentID, this._model.active.getOutputs(this.currentID));
+		},
+		/* Stub to update connections in graph */
+		addQuantity: function(source, destinations){
+		},
+
+		handleEquation: function(equation){
+			var w = registry.byId(this.widgetMap.equation);
+			this.equationEntered = false;
+			w.set('status','');
+			// undo color when new value is entered in the equation box widget
+			w.on("keydown",lang.hitch(this,function(evt){
+				if(evt.keyCode != keys.ENTER){
+					w.set('status','');
+				}
+			}));
+		},
+
+		deleteNode: function(id){
+			//Stub to delete node with id by inturn calling drawmodel.deleteNode in main.js
```

Comment:
Would it be possible if you add TODO here. This get's highlighted in every code editor, and will ensure that we do not forget anything. There are bunch of comments before as well like need to delete all existing outgoing connections on line 391 which is again a TODO.